pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachineUSBDevices/index.vue

```diff
@@ -0,0 +1,229 @@
+import Banner from '@components/Banner/Banner.vue';
+import CompatibilityMatrix from '../CompatibilityMatrix';
+import DeviceList from './DeviceList';
+
+import remove from 'lodash/remove';
+import { get, set } from '@shell/utils/object';
+
+export default {
+  name:       'VirtualMachineUSBDevices',
+  components: {
+    Banner,
+    CompatibilityMatrix,
+    DeviceList,
+    LabeledSelect,
+  },
+  props: {
+    mode: {
+      type:    String,
+      default: _EDIT
+    },
+
+    value: {
+      type:    Object,
+      default: () => {}
+    },
+
+    vm: {
+      type:    Object,
+      default: () => {}
+    }
+  },
+
+  async fetch() {
+    const hash = {
+      devices: this.$store.dispatch('harvester/findAll', { type: HCI.USB_DEVICE }),
+      vms:     this.$store.dispatch(`harvester/findAll`, { type: HCI.VM })
+    };
+
+    const res = await allHash(hash);
+
+    for (const key in res) {
+      this[key] = res[key];
+    }
+
+    this.selectedDevices = (this.value?.domain?.devices?.hostDevices || [])
+      .map(({ name }) => name)
+      .filter(name => this.enabledDevices.find(device => device?.metadata?.name === name));
+  },
+
+  data() {
+    return {
+      deviceSchema:  this.$store.getters['harvester/schemaFor'](HCI.USB_DEVICE),
+      deviceHeaders: [
+        { ...STATE },
+        SIMPLE_NAME,
+      ],
+      devices:         [],
+      vms:             [],
+      selectedDevices: [],
+      showMatrix:      false,
+    };
+  },
+
+  watch: {
+    selectedDevices(neu) {
+      const formatted = neu.map((selectedDevice) => {
+        const deviceCRD = this.enabledDevices.find(device => device.metadata.name === selectedDevice);
+        const deviceName = deviceCRD?.status?.resourceName;
+
+        return {
+          deviceName,
+          name: deviceCRD?.metadata.name,
+        };
+      });
+
+      const devices = [
+        ...this.otherDevices(this.value.domain.devices.hostDevices || []),
+        ...formatted,
+      ];
+
+      set(this.value.domain.devices, 'hostDevices', devices);
+    }
+  },
+
+  computed: {
+    deviceOpts() {
+      const filteredOptions = this.enabledDevices.filter((deviceCRD) => {
+        if (this.selectedDevices.length > 0) {
+          const selectedDevice = this.enabledDevices.find(device => device.metadata.name === this.selectedDevices[0]);
+
+          return !this.devicesInUse[deviceCRD?.metadata.name] && deviceCRD.status.nodeName === selectedDevice?.status.nodeName;
+        }
+
+        return !this.devicesInUse[deviceCRD?.metadata.name];
+      });
+
+      return filteredOptions.map((deviceCRD) => {
+        return {
+          value:        deviceCRD?.metadata.name,
+          label:        deviceCRD?.metadata.name,
+          displayLabel: deviceCRD?.status?.description,
+        };
+      });
+    },
+
+    enabledDevices() {
+      return this.devices.filter((device) => {
+        return device.status.enabled;
+      }) || [];
+    },
```

Comment:
I think we could remove `[]`.
```suggestion
    enabledDevices() {
      return this.devices.filter((device) => {
        return device.status.enabled;
      });
    },
```